### PR TITLE
Implement *WasmAbi for array

### PR DESCRIPTION
**Pull request by c410-f3r**
WIP

Requesting review because this is a blind attempt to implement `*WasmAbi` for arrays (up to 32). Sincerely, I am not aware of the consequences of these changes, let alone the possible missing places required for this feature.

### CODE REVIEW

File: src/convert/slices.rs [hunk 1]
```diff
@@ -271,3 +271,63 @@ if_std! {
...
+            impl<T> IntoWasmAbi for [T; $N]
+            where
+                T: Copy + IntoWasmAbi<Abi = WasmSlice>
+            {
+                type Abi = WasmSlice;
+            
+                #[inline]
+                fn into_abi(self) -> Self::Abi {
+                    WasmSlice {
+                        ptr: self.as_ptr().into_abi(),
```
Comment (alexcrichton) [inline src/convert/slices.rs]: I believe this is a use-after-free, right? `self` is dropped at the end of this function but we're returning a pointer to it?

Comment (c410-f3r) [reply src/convert/slices.rs]: Hehehe. I should have paid more attention instead of simply copying and pasting. Thank you

**Comment by c410-f3r** [comment]
I am sorry for rushing this feature, maybe it would be wiser to ask for mentoring in #122.

After some days struggling to implement the necessary bits, I am definitely stuck  and still wondering what can be done or should be done (Maybe some `walrus` work? Not sure...).

Well, can someone tell the path for this feature? It doesn't need to be a highly detailed report

**Comment by c410-f3r** [comment]
@alexcrichton At the time. there were some concerns about the feasibility of this feature. Has anything changed since then?

**Comment by alexcrichton** [comment]
Unfortunately, no, not much has changed I believe.

**Comment by c410-f3r** [comment]
@alexcrichton Thank you !

With the imminent stabilization of constant generics, it is unfortunate that this feature can't be implemented.
Feel free to ping me if anything changes in the near future so I can continue the work on this PR

**Comment by c410-f3r** [comment]
Sup @alexcrichton 

Is it still not possible to implement this feature?

**Comment by NewDark90** [comment]
I know I need this functionality.

**Pull request closed without merging**